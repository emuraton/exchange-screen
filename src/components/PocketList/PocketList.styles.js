import styled from 'styled-components';

export const Container = styled.div`
  padding: 10px 30px 0 30px;
  color: white;
  background-color: ${props =>
    props.pocketType === 'active'
      ? 'rgb(0, 117, 235, 0.7)'
      : 'rgb(0, 117, 235, 1.1)'};
`;
