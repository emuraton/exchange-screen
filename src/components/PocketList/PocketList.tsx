import React from 'react';
import Carousel from 'nuka-carousel';

import Pocket from '../Pocket/Pocket';

import { Container } from './PocketList.styles';

type IProps = {
  pockets: any;
  inputActiveHandleChange?: Function;
  updateNavigationDispatch: Function;
  pocketType: string;
  inputValue?: string;
  clearInputValues: Function;
};

const PocketList = ({
  pockets,
  inputActiveHandleChange,
  updateNavigationDispatch,
  pocketType,
  inputValue,
  clearInputValues
}: IProps) => {
  const pocketsValues =
    pocketType === 'active'
      ? Object.values(pockets)
      : Object.values(pockets).reverse();

  return (
    <Container pocketType={pocketType}>
      <Carousel
        wrapAround
        afterSlide={slideIndex => {
          updateNavigationDispatch(
            slideIndex,
            pocketType === 'active'
              ? 'activeSelectedPocket'
              : 'balanceSelectedPocket'
          );
          clearInputValues();
        }}
      >
        {pocketsValues.map((pocket: any) => (
          <Pocket
            currency={pocket.currency}
            currencyLabel={pocket.currencyLabel}
            pocketAmount={pocket.amount}
            onChangeHandler={inputActiveHandleChange}
            key={`${
              pocketType === 'active' ? 'Active Pocket' : 'Balance Pocket'
            } - ${pocket.currency}`}
            inputValue={inputValue}
          />
        ))}
      </Carousel>
    </Container>
  );
};

export default PocketList;
