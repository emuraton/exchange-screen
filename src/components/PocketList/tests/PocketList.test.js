import React from 'react';
import { shallow } from 'enzyme';

import PocketList from '../PocketList';

describe('<PocketList />', () => {
  const initialProps = {
    pockets: {
      EUR: { currency: '€', amount: 50.33, currencyLabel: 'EUR' },
      GBP: { currency: '£', amount: 10, currencyLabel: 'GBP' }
    },
    inputActiveHandleChange: jest.fn(),
    updateNavigationDispatch: jest.fn(),
    pocketType: 'active',
    inputValue: '',
    clearInputValues: jest.fn()
  };

  test('should not crash', () => {
    const wrapper = shallow(<PocketList {...initialProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should have 2  ACTIVE pockets in the right order', () => {
    const wrapper = shallow(<PocketList {...initialProps} />);
    const FirstPocket = wrapper.find('Pocket').first();
    expect(FirstPocket.prop('currencyLabel')).toBe('EUR');

    const SecondPocket = wrapper.find('Pocket').at(1);
    expect(SecondPocket.prop('currencyLabel')).toBe('GBP');
  });
});
