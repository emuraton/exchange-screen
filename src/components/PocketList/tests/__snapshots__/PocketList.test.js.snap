// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PocketList /> should not crash 1`] = `
<styled.div
  pocketType="active"
>
  <Carousel
    afterSlide={[Function]}
    autoGenerateStyleTag={true}
    autoplay={false}
    autoplayInterval={3000}
    autoplayReverse={false}
    beforeSlide={[Function]}
    cellAlign="left"
    cellSpacing={0}
    disableAnimation={false}
    disableEdgeSwiping={false}
    dragging={true}
    easing="easeCircleOut"
    edgeEasing="easeElasticOut"
    enableKeyboardControls={false}
    frameOverflow="hidden"
    framePadding="0px"
    height="inherit"
    heightMode="max"
    onDragStart={[Function]}
    onResize={[Function]}
    pauseOnHover={true}
    renderAnnounceSlideMessage={[Function]}
    renderBottomCenterControls={[Function]}
    renderCenterLeftControls={[Function]}
    renderCenterRightControls={[Function]}
    slideIndex={0}
    slideListMargin={10}
    slideOffset={25}
    slideWidth={1}
    slidesToScroll={1}
    slidesToShow={1}
    speed={500}
    style={Object {}}
    swiping={true}
    transitionMode="scroll"
    vertical={false}
    width="100%"
    withoutControls={false}
    wrapAround={true}
  >
    <Pocket
      currency="€"
      currencyLabel="EUR"
      inputValue=""
      key="Active Pocket - €"
      onChangeHandler={[MockFunction]}
      pocketAmount={50.33}
    />
    <Pocket
      currency="£"
      currencyLabel="GBP"
      inputValue=""
      key="Active Pocket - £"
      onChangeHandler={[MockFunction]}
      pocketAmount={10}
    />
  </Carousel>
</styled.div>
`;
